lib/documents.js

```diff
@@ -29,6 +29,10 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
     collection: 'tokenization-registration',
     fields: {'registration.externalIdHash': 1, 'registration.documentHash': 1},
     options: {unique: true, background: false}
+  }, {
+    collection: 'tokenization-registration',
+    fields: {'registration.internalId': 1},
+    options: {unique: true, background: false}
```

Comment:
So this is tricky.

We previously determined that we couldn't add this index for performance reasons. Sharding is on `externalIdHash` and this does not include it and this would cause scatter-gather queries (and our query cannot include `externalIdHash` in `getRegistration`). This shouldn't be a problem on its own (I don't think), but, IIRC, this index creates insertion performance problems. We should re-evaluate this unless someone else better remembers the details.

I thought this was better documented in the `getRegistration` call to indicate that it will always be an unindexed slow call -- but apparently not. Regardless of whether we add this index, we should add documentation to that function to indicate that it is a low performance (potentially very low) method -- either it will do a full doc scan or it will do scatter-gather (based on whether we can add this index).

If we end up adding this index we should also add a comment here indicating that it doesn't include the shard key so it will result in scatter-gather queries (and we know that and it's ok). If we opt to have no index here we should also indicate why we don't have one.